src/libs/sdlcd/linux/SDL_syscdrom.c

```diff
@@ -0,0 +1,574 @@
+static int SDL_SYS_CDPause(SDL_CD *cdrom);
+static int SDL_SYS_CDResume(SDL_CD *cdrom);
+static int SDL_SYS_CDStop(SDL_CD *cdrom);
+static int SDL_SYS_CDEject(SDL_CD *cdrom);
+static void SDL_SYS_CDClose(SDL_CD *cdrom);
+
+/* Some ioctl() errno values which occur when the tray is empty */
+#ifndef ENOMEDIUM
+#define ENOMEDIUM ENOENT
+#endif
+#define ERRNO_TRAYEMPTY(errno)	\
+	((errno == EIO)    || (errno == ENOENT) || \
+	 (errno == EINVAL) || (errno == ENOMEDIUM))
+
+/* Check a drive to see if it is a CD-ROM */
+static int CheckDrive(char *drive, char *mnttype, struct stat *stbuf)
+{
+	int is_cd, cdfd;
+	struct cdrom_subchnl info;
+
+	/* If it doesn't exist, return -1 */
+	if ( drive == NULL )
+		return(-1);
+	cdfd = open(drive, (O_RDONLY|O_NONBLOCK), 0);
+	if ( stat(drive, stbuf) < 0 ) {
+		if ( cdfd >= 0 )
+			close(cdfd);
+		return(-1);
+    }
+
+	/* If it does exist, verify that it's an available CD-ROM */
+	is_cd = 0;
+	if ( S_ISCHR(stbuf->st_mode) || S_ISBLK(stbuf->st_mode) ) {
+		if ( cdfd >= 0 ) {
+			info.cdsc_format = CDROM_MSF;
+			/* Under Linux, EIO occurs when a disk is not present.
+			 */
+			if ( (ioctl(cdfd, CDROMSUBCHNL, &info) == 0) ||
+						ERRNO_TRAYEMPTY(errno) ) {
+				is_cd = 1;
+			}
+		}
+#ifdef USE_MNTENT
+		/* Even if we can't read it, it might be mounted */
+		else if ( mnttype && (SDL_strcmp(mnttype, MNTTYPE_CDROM) == 0) ) {
+			is_cd = 1;
+		}
+#endif
+	}
+	if ( cdfd >= 0 )
+		close(cdfd);
+	return(is_cd);
+}
+
+/* Add a CD-ROM drive to our list of valid drives */
+static void AddDrive(char *drive, struct stat *stbuf)
+{
+	int i;
+
+	if ( SDL_numcds < MAX_DRIVES ) {
+		/* Check to make sure it's not already in our list.
+	 	   This can happen when we see a drive via symbolic link.
+		 */
+		for ( i=0; i<SDL_numcds; ++i ) {
+			if ( stbuf->st_rdev == SDL_cdmode[i] ) {
+#ifdef DEBUG_CDROM
+  fprintf(stderr, "Duplicate drive detected: %s == %s\n", drive, SDL_cdlist[i]);
+#endif
+				return;
+			}
+		}
+
+		/* Add this drive to our list */
+		i = SDL_numcds;
+		SDL_cdlist[i] = SDL_strdup(drive);
+		if ( SDL_cdlist[i] == NULL ) {
+			SDL_OutOfMemory();
+			return;
+		}
+		SDL_cdmode[i] = stbuf->st_rdev;
+		++SDL_numcds;
+#ifdef DEBUG_CDROM
+  fprintf(stderr, "Added CD-ROM drive: %s\n", drive);
+#endif
+	}
+}
+
+#ifdef USE_MNTENT
+#define STRMAXLEN 300
+static void CheckMounts(const char *mtab)
+{
+	FILE *mntfp;
+	struct mntent *mntent;
+	struct stat stbuf;
+
+	mntfp = setmntent(mtab, "r");
+	if ( mntfp != NULL ) {
+		char *tmp;
+		size_t mnt_type_len;
+		size_t mnt_dev_len;
+		char mnt_type_str[STRMAXLEN+1];
+		char mnt_dev_str[STRMAXLEN+1];
+		char *mnt_type = mnt_type_str;
+		char *mnt_dev = mnt_dev_str;
+
+		while ( (mntent=getmntent(mntfp)) != NULL ) {
+			*mnt_type = '\0';
+			*mnt_dev = '\0';
+			mnt_type_len = SDL_strlen(mntent->mnt_type) + 1;
+			if (mnt_type_len>STRMAXLEN)
+				strncpy(mnt_type_str, mntent->mnt_type, STRMAXLEN);
+			else
+				strcpy(mnt_type_str, mntent->mnt_type);
```

Comment:
Unbounded C-string copy defect caught by Coverity:

![2022-02-14_14-51_1](https://user-images.githubusercontent.com/1557255/153960634-1dac9f6f-02df-43e9-9caf-86df4eb786fd.png)

Reply:
@kcgen If you look at line 215, what it does is checking the length of mnt_type_str. Not sure why it is still being reported.